### ISSUE TITLE
Using DLL - GImGui was a nullpointer ImGui::Begin

### ISSUE THREAD

**Issue by silvertakana** [ocornut/imgui#5248]
**Version/Branch of Dear ImGui:**

Version: v1.88 WIP
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_glfw.cpp + imgui_impl_opengl3.cpp
Compiler: vs2022 v17.2.0 Preview 4.0
Operating System: Windows 11

**My Issue/Question:**

when trying to create an ImGui window, my program crashed at the function `ImGui::FindWindowByID` throwing `Exception thrown: read access violation.
GImGui was nullptr.` I'm guessing I might have forgotten to initialize something. can anyone tell me what that something could be?

**Screenshots/Video**
here Is the call stack
![image](https://user-images.githubusercontent.com/62156548/165452339-4ccc63de-1115-4af6-b33f-b0de3bdaf483.png)

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```c++
ImGui::Begin("Test");
ImGui::Text("Hello World");
ImGui::End();
```

**Comment by PathogenDavid** [comment]
> an anyone tell me what that something could be?

Literally everything. `GImGui` being null indicates Dear ImGui isn't initialized at all.

See the examples for how to initialize things:

https://github.com/ocornut/imgui/blob/e54db4ee708b99e24348f31847e8032921913989/examples/example_glfw_opengl3/main.cpp#L64-L77

**Comment by silvertakana** [comment]
I've already initialized it. here is the proof
![image](https://user-images.githubusercontent.com/62156548/165657030-dfd0a643-6748-438a-a0ee-7581b2a0bcc8.png)
what else could be wrong?

**Comment by silvertakana** [comment]
Turns out there were two GImGui variables, the original one that I initialized and the other one that was nullptr when I call `ImGui::Begin`. so I tried storing the context in a void variable and setting it back in the function.
```c++
	void OnImGuiRender() override
	{
		ImGui::SetCurrentContext((ImGuiContext*)ImGuiCtx);
		ImGui::Begin("Test");
		ImGui::Text("Hello world!");
		ImGui::End();
	}
```
It did work and displayed a nice window. But I feel like it was just a bit hacky. is there a way around it? and why was it a different variable?
